nipype/interfaces/freesurfer/petsurfer.py

```diff
@@ -0,0 +1,652 @@
+    --seg gtmseg.mgz'
+    """
+
+    _cmd = "mri_gtmpvc"
+    input_spec = GTMPVCInputSpec
+    output_spec = GTMPVCOutputSpec
+
+    def _list_outputs(self):
+        outputs = self.output_spec().get()
+        # Get the top-level output directory
+        if not isdefined(self.inputs.pvc_dir):
+            pvcdir = os.getcwd()
+        else:
+            pvcdir = os.path.abspath(self.inputs.pvc_dir)
+        outputs["pvc_dir"] = pvcdir
+
+        # Assign the output files that always get created
+        outputs["ref_file"] = os.path.join(pvcdir, "km.ref.tac.dat")
+        outputs["hb_nifti"] = os.path.join(pvcdir, "km.hb.tac.nii.gz")
+        outputs["hb_dat"] = os.path.join(pvcdir, "km.hb.tac.dat")
+        outputs["nopvc_file"] = os.path.join(pvcdir, "nopvc.nii.gz")
+        outputs["gtm_file"] = os.path.join(pvcdir, "gtm.nii.gz")
+        outputs["gtm_stats"] = os.path.join(pvcdir, "gtm.stats.dat")
+        outputs["reg_pet2anat"] = os.path.join(pvcdir, "aux/bbpet2anat.lta")
+
+        # Assign the conditional outputs
+        if isdefined(self.inputs.save_input) and self.inputs.save_input:
+            outputs["input_file"] = os.path.join(pvcdir, "input.nii.gz")
+        if isdefined(self.inputs.save_yhat0) and self.inputs.save_yhat0:
+            outputs["yhat0"] = os.path.join(pvcdir, "yhat0.nii.gz")
+        if isdefined(self.inputs.save_yhat) and self.inputs.save_yhat:
+            outputs["yhat"] = os.path.join(pvcdir, "yhat.nii.gz")
+        if isdefined(self.inputs.save_yhat_full_fov) and self.inputs.save_yhat_full_fov:
+            outputs["yhat_full_fov"] = os.path.join(pvcdir, "yhat_full_fov.nii.gz")
+        if (
+            isdefined(self.inputs.save_yhat_with_noise)
+            and self.inputs.save_yhat_with_noise
+        ):
+            outputs["yhat_with_noise"] = os.path.join(pvcdir, "yhat_with_noise.nii.gz")
+        if isdefined(self.inputs.mgx) and self.inputs.mgx:
+            outputs["mgx_ctxgm"] = os.path.join(pvcdir, "mgx.ctxgm.nii.gz")
+            outputs["mgx_subctxgm"] = os.path.join(pvcdir, "mgx.subctxgm.nii.gz")
+            outputs["mgx_gm"] = os.path.join(pvcdir, "mgx.gm.nii.gz")
+        if isdefined(self.inputs.rbv) and self.inputs.rbv:
+            outputs["rbv"] = os.path.join(pvcdir, "rbv.nii.gz")
+        if isdefined(self.inputs.opt) and self.inputs.opt:
+            outputs["opt_params"] = os.path.join(pvcdir, "aux/opt.params.dat")
+
+        return outputs
+
+    def _format_arg(self, name, spec, value):
+        return super(GTMPVC, self)._format_arg(name, spec, value)
+
+
+class MRTMInputSpec(GLMFitInputSpec):
+
+    mrtm1 = InputMultiPath(
+        traits.Tuple(
+            File(exists=True, mandatory=True), File(exists=True, mandatory=True)
+        ),
+        argstr="--mrtm1 %s %s...",
+        desc="RefTac TimeSec : perform MRTM1 kinetic modeling",
+    )
+
+
+class MRTMOutputSpec(GLMFitInputSpec):
+
+    k2p = File(desc="estimate of k2p parameter")
+
+
+class MRTM(GLMFit):
```

Comment:
These subclasses of `GLMFit` seem to be doing two things:

1) Making the inputs for specific use cases mandatory.
2) Adding outputs that depend on these inputs.

I think we should handle the outputs in `GLMFit` itself, and also go ahead and provide examples for these use cases in the main docstring. Then I would pare these down to a quite bare:

```Python
class MRTMInputSpec(GLMFitInputSpec):
    mrtm1 = InputMultiPath(
        traits.Tuple(File(exists=True), File(exists=True)),
        mandatory=True,
        argstr="--mrtm1 %s %s...",
        desc="RefTac TimeSec : perform MRTM1 kinetic modeling",
    )

class MRTM(GLMFit):
    """ Run FreeSurfer mri_glmfit to perform MRTM1 kinetic modeling.

    Use this interface to make ``mrtm1`` a required input.
    See :py:class:`~nipype.interfaces.freesurfer.modeling.GLMFit` for details.
    """
    input_spec = MRTMInputSpec
```

Here's a proposed diff to implement MRTM1 in `GLMFit`:

```Diff
diff --git a/nipype/interfaces/freesurfer/model.py b/nipype/interfaces/freesurfer/model.py
index dbde69dd0..2d0ba08a6 100644
--- a/nipype/interfaces/freesurfer/model.py
+++ b/nipype/interfaces/freesurfer/model.py
@@ -404,7 +404,7 @@ class GLMFitInputSpec(FSTraitedSpec):
     mrtm1 = InputMultiPath(
         traits.Tuple(File(exists=True), File(exists=True)),
         argstr="--mrtm1 %s %s...",
-        desc="Reference time activity curve and frame times",
+        desc="Perform MRTM1 kinetic modeling, given reference time activity curve and frame times",
     )
     mrtm2 = InputMultiPath(
         traits.Tuple(File(exists=True), File(exists=True), traits.Float),
@@ -463,6 +463,7 @@ class GLMFitOutputSpec(TraitedSpec):
     frame_eigenvectors = File(desc="matrix of frame eigenvectors from residual PCA")
     singular_values = File(desc="matrix singular values from residual PCA")
     svd_stats_file = File(desc="text file summarizing the residual PCA")
+    k2p_file = File(desc="estimate of k2 prime parameter")
 
 
 class GLMFit(FSCommand):
@@ -473,9 +474,18 @@ class GLMFit(FSCommand):
     >>> glmfit = GLMFit()
     >>> glmfit.inputs.in_file = 'functional.nii'
     >>> glmfit.inputs.one_sample = True
-    >>> glmfit.cmdline == 'mri_glmfit --glmdir %s --y functional.nii --osgm'%os.getcwd()
-    True
-
+    >>> glmfit.cmdline  # doctest: +ELLIPSIS
+    'mri_glmfit --glmdir ... --y functional.nii --osgm'
+
+    ``mri_glmfit`` can perform MRTM1 kinetic modeling, which will generate a ``k2p_file``
+
+    >>> mrtm = GLMFit(mrtm1=[('ref_tac.dat', 'timing.dat')])
+    >>> mrtm.inputs.in_file = 'tac.nii'
+    >>> mrtm.inputs.glm_dir = 'mrtm'
+    >>> mrtm.cmdline
+    'mri_glmfit --glmdir mrtm --y tac.nii --mrtm1 ref_tac.dat timing.dat'
+    >>> mrtm._list_outputs()['k2p_file']  # doctest: +ELLIPSIS
+    '.../mrtm/k2prime.dat'
     """
 
     _cmd = "mri_glmfit"
@@ -509,10 +519,12 @@ class GLMFit(FSCommand):
             outputs["error_file"] = os.path.join(glmdir, "eres.mgh")
         if isdefined(self.inputs.save_estimate) and self.inputs.save_estimate:
             outputs["estimate_file"] = os.path.join(glmdir, "yhat.mgh")
+        if self.inputs.mrtm1:
+            outputs["k2p_file"] = os.path.join(glmdir, "k2prime.dat")
 
         # Get the contrast directory name(s)
+        contrasts = []
         if isdefined(self.inputs.contrast):
-            contrasts = []
             for c in self.inputs.contrast:
                 if split_filename(c)[2] in [".mat", ".dat", ".mtx", ".con"]:
                     contrasts.append(split_filename(c)[1])
```

WDYT?

Reply:
Solved via [d3efa06](https://github.com/nipy/nipype/pull/3437/commits/d3efa065294a3d86f2639c1132a70a18cae2e631)